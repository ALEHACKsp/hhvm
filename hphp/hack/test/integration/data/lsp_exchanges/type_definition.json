[
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "initialize",
  "params":
    {
      "initializationOptions":{},
      "processId":null,
      "rootPath":"${root_path}",
      "capabilities":{}
    }
},
{
  "jsonrpc": "2.0",
  "method": "$test/waitForHhServerReady",
  "params": {}
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
     "textDocument": {
       "uri":"${php_file_uri}",
       "languageId": "hack",
       "version":1,
       "text":"${php_file}"
     }
   }
},
{
  "comment": "Conditional Type Definition of HH or II",
  "jsonrpc": "2.0",
  "method": "textDocument/typeDefinition",
  "id": 2,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 32, "character": 2}
   }
},
{
  "comment": "Standard Class Definition",
  "jsonrpc": "2.0",
  "method": "textDocument/typeDefinition",
  "id": 3,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 40, "character": 2}
   }
},
{
  "comment": "Class Type Definition with Casting",
  "jsonrpc": "2.0",
  "method": "textDocument/typeDefinition",
  "id": 4,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 41, "character": 2}
   }
},
{
  "comment": "Primitive Type Definition",
  "jsonrpc": "2.0",
  "method": "textDocument/typeDefinition",
  "id": 5,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 42, "character": 2}
   }
},
{
  "comment": "Function Return Type Definition",
  "jsonrpc": "2.0",
  "method": "textDocument/typeDefinition",
  "id": 6,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 43, "character": 2}
   }
},
{
  "comment": "Function definition with primitive return type",
  "jsonrpc": "2.0",
  "method": "textDocument/typeDefinition",
  "id": 7,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 44, "character": 2}
   }
},
{
  "jsonrpc": "2.0",
  "id": 99,
  "method": "shutdown",
  "params": {}
}
]
