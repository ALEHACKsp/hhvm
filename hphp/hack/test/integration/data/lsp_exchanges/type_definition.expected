[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "typeDefinitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": [
      {
        "uri": "file://${root_path}/type_definition.php",
        "range": {
          "start": {
            "line": 2,
            "character": 6
          },
          "end": {
            "line": 2,
            "character": 8
          }
        },
        "title": "\\HH"
      },
      {
        "uri": "file://${root_path}/type_definition.php",
        "range": {
          "start": {
            "line": 12,
            "character": 6
          },
          "end": {
            "line": 12,
            "character": 8
          }
        },
        "title": "\\LL"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": [
      {
        "uri": "file://${root_path}/type_definition.php",
        "range": {
          "start": {
            "line": 2,
            "character": 6
          },
          "end": {
            "line": 2,
            "character": 8
          }
        },
        "title": "\\HH"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 4,
    "result": [
      {
        "uri": "file://${root_path}/type_definition.php",
        "range": {
          "start": {
            "line": 2,
            "character": 6
          },
          "end": {
            "line": 2,
            "character": 8
          }
        },
        "title": "\\HH"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 5,
    "result": []
  },
  {
    "jsonrpc": "2.0",
    "id": 6,
    "result": [
      {
        "uri": "file://${root_path}/type_definition.php",
        "range": {
          "start": {
            "line": 12,
            "character": 6
          },
          "end": {
            "line": 12,
            "character": 8
          }
        },
        "title": "\\LL"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 7,
    "result": [
      {
        "uri": "file://${root_path}/type_definition.php",
        "range": {
          "start": {
            "line": 22,
            "character": 9
          },
          "end": {
            "line": 22,
            "character": 29
          }
        },
        "title": "(function(): int)"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 99,
    "result": null
  }
]
